### PR TITLE
feat: support window function row_number

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: docs/en_US/sqls/functions/window_functions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Window Functions
```
Comment (ngjaying) [inline docs/en_US/sqls/functions/window_functions.md:1]: Please also add a link in the parent page.

File: docs/en_US/sqls/functions/window_functions.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+A window function performs a calculation across a set of table rows that are somehow related to the current row. This is comparable to the type of calculation that can be done with an aggregate function.
```
Comment (ngjaying) [inline docs/en_US/sqls/functions/window_functions.md:3]: Add a notice that this kind of functions can only be used in the select clause

File: internal/topo/planner/planner.go [hunk 3]
```diff
@@ -463,6 +465,14 @@ func createLogicalPlan(stmt *ast.SelectStatement, opt *api.RuleOption, store kv.
 		p.SetChildren(children)
 		children = []LogicalPlan{p}
 	}
+	windowFuncFields, windowFuncsNames := extractWindowFuncFields(stmt)
```
Comment (ngjaying) [inline internal/topo/planner/planner.go:468]: Do we support to use window function in order by? Looks like not supported in the planner.

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: internal/topo/planner/planner.go (hunk 3 above)
Comment (Yisaer) [reply internal/topo/planner/planner.go:468]: we will support it later.

**Review by ngjaying — APPROVED** [review]

**Pull request merged**